core/src/main/java/org/apache/calcite/runtime/JsonFunctions.java

```diff
@@ -606,41 +608,160 @@ public static String jsonKeys(JsonPathContext context) {
     return jsonize(list);
   }
 
-  public static String jsonRemove(String input, String... pathSpecs) {
-    return jsonRemove(jsonValueExpression(input), pathSpecs);
+  public static Integer jsonStorageSize(String input) {
+    return jsonStorageSize(jsonValueExpression(input));
   }
 
-  public static String jsonRemove(JsonValueContext input, String... pathSpecs) {
+  public static Integer jsonStorageSize(JsonValueContext input) {
     try {
-      DocumentContext ctx = JsonPath.parse(input.obj,
-          Configuration
-              .builder()
-              .options(Option.SUPPRESS_EXCEPTIONS)
-              .jsonProvider(JSON_PATH_JSON_PROVIDER)
-              .mappingProvider(JSON_PATH_MAPPING_PROVIDER)
-              .build());
-      for (String pathSpec : pathSpecs) {
+      return JSON_PATH_JSON_PROVIDER.getObjectMapper()
+          .writeValueAsBytes(input.obj).length;
+    } catch (Exception e) {
+      throw RESOURCE.invalidInputForJsonStorageSize(Objects.toString(input.obj)).ex();
+    }
+  }
+
+  private static String jsonModify(JsonValueContext jsonDoc, String type, Object... kvs) {
+    JsonModifyMode jsonPairsType = JsonModifyMode.getType(type);
+    if (jsonPairsType != JsonModifyMode.REMOVE) {
+        assert kvs.length % 2 == 0;
+    }
+    String result = "";
```

Comment:
If I am not wrong. Per operator definition return types of all these methods are nullable. Please either use null or throw error rather than returning empty string when something unexpected occurs.